### PR TITLE
fix: add `__str__` to `ApplicationCommand`

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Adds `__str__` to `ApplicationCommand`, previously only implemented for `SlashCommand` while the other types would fall back to `__repr__`.
This is purely a cosmetic fix, and prevents this from happening with command sync debugging:
```py
COMMANDS IN 1234
===============================
| Update is required: False
| To upsert:
|     -
| To edit:
|     -
| To delete:
|     -
| No changes:
|     <SlashCommand name='localized_command'>
|     <SlashCommand name='localized_top_level'>
|     <MessageCommand name='Localized Reverse' default_permission=True>
```
(note the `default_permission` being shown)

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/app_commands.py [hunk 1]
```diff
@@ -357,6 +357,9 @@ def __repr__(self) -> str:
         attrs = " ".join(f"{key}={getattr(self, key)!r}" for key in self.__repr_info__)
         return f"<{type(self).__name__} {attrs}>"
 
+    def __str__(self) -> str:
+        return f"<{type(self).__name__} name={self.name!r}>"
```
Comment (onerandomusername) [inline disnake/app_commands.py:361]: if we're using ``__str__``, why are we making this like a repr? This should be just the command name, as it is *not* a repr.

Comment (onerandomusername) [reply disnake/app_commands.py:361]: if we want something specific for debug syncing printing, we should do something different.

Comment (shiftinv) [reply disnake/app_commands.py:361]: You wouldn't be able to tell apart the different types of commands if it just returned the command name, and knowing the type of command seems like something you'd want for debugging.
We could just remove `__str__` here entirely, and do this in the sync debug logging part of the code?

Comment (shiftinv) [reply disnake/app_commands.py]: https://github.com/DisnakeDev/disnake/pull/478/commits/e8bf2c03ee0c5473e98c40edf1d7c2eabc378b85

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**